lib/auth/auth.go

```diff
@@ -906,9 +911,25 @@ func (a *Server) ExtendWebSession(user, prevSessionID, accessRequestID string, i
 		}
 	}
 
+	if req.SwitchBackDefaults != nil {
+		defaultRoles := req.SwitchBackDefaults.Roles
+		defaultExpiry := req.SwitchBackDefaults.Expires
+
+		if err := utils.StringSliceSubset(roles, defaultRoles); err != nil {
+			return nil, trace.Wrap(err)
```

Comment:
The function's error message will be pretty generic - we should amend the message to include the relevant bits about role mismatch.